src/Core/Settings.cpp

```diff
@@ -4878,6 +4878,9 @@ Wait time to lock cache for space reservation in filesystem cache
 )", 0) \
     DECLARE(Bool, filesystem_cache_prefer_bigger_buffer_size, true, R"(
 Prefer bigger buffer size if filesystem cache is enabled to avoid writing small file segments which deteriorate cache performance. On the other hand, enabling this setting might increase memory usage.
+)", 0) \
+    DECLARE(UInt64, filesystem_cache_boundary_alignment, 0, R"(
+Filesystem cache boundary alignment. This setting is applied only for non-disk read. Value 0 means no alignment.
```

Comment:
As written, it is only for cache for remote table engines/functions, not for remote disks.
Added because it drastically improves read performance for formats like Parquet.